src/kademlia/dht_storage.cpp

```diff
@@ -346,6 +345,7 @@ namespace
 		}
 
 		void put_immutable_item(sha1_hash const& target
+			, sha1_hash const& our_id
```

Comment:
hm.. it's possible that there may be some clever hack, but I kind of doubt it. I think it would have to come down to calculating the distance to multiple node IDs and take the closest one as the "priority".

distance_exp() literally just takes a node IDs, XORs it with our node ID and then returns the xor distance between them (160 - number_of_leading_zeros)